contrib/babelfishpg_tsql/src/multidb.c

```diff
@@ -667,6 +676,125 @@ rewrite_relation_walker(Node *node, void *context)
 		return raw_expression_tree_walker(node, rewrite_relation_walker, context);			
 }
 
+/*
+ * select_json_modify takes in a select statement
+ * If the target is json_modify and the from clause is for json we change the function name
+ * to json_modify_no_escape which handles this case properly
+ * Otherwise we do not change anything
+ */
+void select_json_modify(SelectStmt* stmt)
+{
+	List* targList = stmt->targetList;
+	List* fromList = stmt->fromClause;
+	if(list_length(targList) != 0 && list_length(fromList) != 0)
+	{
+		Node* n = linitial(targList);
+		Node* n_from = linitial(fromList);
+		if(IsA(n, ResTarget) && IsA(n_from, RangeSubselect))
+		{
+			ResTarget* rt = (ResTarget*) n;
+			RangeSubselect* rs = (RangeSubselect*) n_from;
+			if(IsA(rt->val, FuncCall) && IsA(rs->subquery, SelectStmt))
+			{
+				FuncCall* json_mod_fc = (FuncCall*) rt->val;
+				SelectStmt* from_sel_stmt = (SelectStmt*) rs->subquery;
+				rewrite_plain_name(json_mod_fc->funcname);
+				if(is_json_modify(json_mod_fc->funcname))
+				{
+					A_Const *escape = makeNode(A_Const);
+					escape->val.boolval.type = T_Boolean;
+					escape->val.boolval.boolval = is_select_for_json(from_sel_stmt);
+					escape->location = -1;
+					json_mod_fc->args = list_delete_last(json_mod_fc->args);
```

Comment:
Is it not possible to modify the existing `escape` value instead of deleting and adding a new node?